src/util.cpp

```diff
@@ -435,11 +435,116 @@ static void InterpretNegativeSetting(std::string& strKey, std::string& strValue)
     }
 }
 
+/** Internal helper functions for ArgsManager */
+class ArgsManagerHelper {
+public:
+    inline static bool UseDefaultSection(const ArgsManager& am, const std::string& strArg)
+    {
+        if (am.m_strSection == CBaseChainParams::MAIN)
+            return true;
+        else if (am.m_setSectionOnlyArgs.count(strArg) == 0)
+            return true;
+        else
+            return false;
+    }
+
+    typedef std::map<std::string, std::vector<std::string>> map_args;
+
+    /** Convert regular argument into the section-specific setting */
+    inline static std::string SectionArg(const ArgsManager& am, const std::string& strArg)
+    {
+        assert(strArg.length() > 1 && strArg[0] == '-');
```

Comment:
This assert can be hit. Try running `feature_pruning.py` for example:

```
bitcoind: util.cpp:456: static std::__cxx11::string ArgsManagerHelper::SectionArg(const ArgsManager&, const string&): Assertion `strArg.length() > 1 && strArg[0] == '-'' failed.
```

Reply:
I believe this is catching a legitimate bug in the caller; confirming at the minute.